cmake/ot_libs_substitute.cmake

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+include(${NRFXLIB_DIR}/common.cmake)
+
+# Copy new OpenThread libraries, headers, text file with the configuration
+# to the nrfxlib openthread folder.
+function(ot_libs_substitute)
+  find_package(Git QUIET)
+  if(GIT_FOUND)
+    execute_process(
+      COMMAND           ${GIT_EXECUTABLE} rev-parse HEAD
+      WORKING_DIRECTORY ${openthread_SOURCE_DIR}
+      RESULT_VARIABLE   git_return
+      OUTPUT_VARIABLE   openthread_git_hash
+      )
+
+    list(APPEND OPENTHREAD_SETTINGS "OpenThread_commit=${openthread_git_hash}\n\n")
+  endif()
+
+  # Store all OT related variables
+  get_cmake_property(_variableNames VARIABLES)
+  list (SORT _variableNames)
+  foreach (_variableName ${_variableNames})
+    if("${_variableName}" MATCHES "^CONFIG_OPENTHREAD_.*|^OT_.*")
+      list(APPEND OPENTHREAD_SETTINGS "${_variableName}=${${_variableName}}\n")
+    endif()
+  endforeach()
+
+  set(OPENTHREAD_CONFIG_FILE
+    "${PROJECT_BINARY_DIR}/openthread_lib_configuration.txt")
+  FILE(WRITE ${OPENTHREAD_CONFIG_FILE} ${OPENTHREAD_SETTINGS})
+
+  set(OPENTHREAD_SRC_DIR "${PROJECT_BINARY_DIR}/modules/openthread/build/src")
+  set(OPENTHREAD_LIBS
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-radio.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-rcp.a"
+  )
+
+  nrfxlib_calculate_lib_path(lib_path)
+
+  foreach(lib IN LISTS OPENTHREAD_LIBS)
+    add_custom_command(
+      TARGET app
+      COMMENT "Copying OpenThread library: ${lib}."
+      PRE_LINK
```

Comment:
We should not copy / overwrite files under version control as part of building the application.

This must be a deliberate action taken by the user.

CMake has a dedicated install target for such operations (which NCS doesn't use, as it make no sense for embedded systems).
But we should follow the principle of having a dedicated target to call when installing the libs.
`ninja install_openthread_libraries`

Reply:
Agree. Will follow this approach.